terraform/README.md

```diff
@@ -0,0 +1,153 @@
+# Cantaloupe Terraform Deployment
+
+## Prerequisites for using docker-cantaloupe terraform
+In order to deploy successfully, you'll need the following prerequisites:
+* Terraform Enterprise (Free/Paid) Account
+  * Create an organization or use your own account name as your organization
+* Terraform Enterprise (Free Tier) Token
+  * Generate a user token from: https://app.terraform.io/app/settings/tokens
+* Create a `backend.hcl` file by replacing the values found in the `backend.hcl.sample` file
+* The [Terraform software](https://www.terraform.io/downloads.html) installed on your machine.
+* Supply a prefix to name your AWS resources with. Usually an alias similar to your username will suffice
+  * `ephemeral_app_name = "joebruin"`
+* VPC Setup
+  * This setup assumes you have no VPC set up. We want the ephemeral environments isolated from existing resources
+  * The default setup will create a `10.0.0.0/16` VPC network and a `10.0.1.0/24` subnet allocated to the created region. If you're not sure what to do here, its suggested you leave this alone unless you have conflicting IP spaces in your AWS environment
+  * OPTIONAL: If you want to override and of the existing VPC configurations, you'll need to supply this information. Note that you'll need to 
+    * The following example will create a `10.1.0.0/16` VPC network and 2 subnets in different AZ zones in the same region `10.1.1.0/24` and `10.1.2.0/24`:
+    * `vpc_cidr_block    = 10.1.0.0/16`
+    * `subnet_init_value = 1`
+    * `subnet_count      = 2`
+* Security Group Configuration
+  * Supply the port in which your application runs on. There is no port forwarding allowed in Fargate setups
+    * `ingress_port = [8181, 8182]`
+  * Supply what networks and/or IPs you want to allow access to your Fargate container
+    * `ingress_allowed = [0.0.0.0/16]`
+  * OPTIONAL: If you have more restrictive needs and only want to allow AWS resources to access your containers, you can define the IDs of the resources in the following:
+    * `sg_groups = ["${aws_lb.main.id}"]`
+* Private Image Registry Access
+  * If you are accessing an image from a private registry, you'll need to store the credentials in AWS Secrets Manager. After storing the secrets, you'll need to login to the IAM console, retrieve the ARN and supply it to this terraform deployment
+    * `dockerhub_credentials_secrets_arn = "arn:aws:secretsmanager:us-west-2:0123456789:secret:dockerhubauth-example"`
```

Comment:
This looks like one config but in systems manager there are two: dockerhub username and dockerhub password. In talking with you, you mention secrets manager is something different. It would be nice to have more detail here or even better a terraform config to create this resource for me.

Reply:
I left the documentation as is. I'll create another ticket in JIRA for adding terraform-created secrets in a future release.